.github/workflows/build.yml

```diff
@@ -8,12 +8,12 @@ jobs:
     strategy:
       matrix:
         operating-system: [ubuntu-latest]
-        php-versions: ['7.2', '7.3', '7.4', '8.0', '8.1']
+        php-versions: ['7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3-dev']
```

Comment:
`8.3` is the unstable version and I think we don't test the development version.

Reply:
> It seems that the CI checking is failed during the `PHP 8.2` tests running.
> 
> Is the internal `PHP 8.2` error?

I also get the failed test in my local machine:

```sh
$ php -v
PHP 8.2.0 (cli) (built: Dec 10 2022 10:52:23) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.0, Copyright (c), by Zend Technologies
$ vendor/bin/phpunit --exclude-group mayignore
vendor/bin/phpunit --exclude-group mayignore
PHPUnit 8.5.31 by Sebastian Bergmann and contributors.

................SSSSSSS..SSS......................W............  63 / 306 ( 20%)
......................Segmentation fault (core dumped)
```

Reply:
After digging the issue, it seems that the enabling VCR with [php-vcr](https://github.com/php-vcr/php-vcr) is problematic.

Reply:
After removing related VCR enabling/disabling codes, the test suites are worked successfully.

Reply:
I also describe the https://github.com/php-vcr/php-vcr/issues/373 on this repository.

Reply:
At this moment, I think we don't support PHP `8.2` version.

Please remove the PHP `8.2` test during GitHub action.

Reply:
@peter279k I we are mixing up different issues here.

>At this moment, I think we don't support the PHP 8.2 version.

IMO this is not because a version is run in the CI that it means it's supported. I think it is beneficial to add it as if you want to work on PHP 8.2 support, then having it as part of the CI is a nice to have.

I also think I PHPBrew doesn't support PHP 8.2 & 8.3, then it should convey it from the `composer.json`. The current constraint `^8.0` indicates otherwise. Would you be fine with fixing this?

Last but not least: I see that there is no required check in the pipeline. I recommend to enable them: Settings -> Branches -> Add a rule or edit the rule for the `master` branch and:
- tick  "Require status checks to pass before merging"
- look for status checks to check

This way you have a distinction between a build that is required, and one that is purely informative (e.g. 8.2 or 8.3 compatibility)

Reply:
Thanks for your reply. I think we can keep the PR until these versions are supported and compatible.

The PHP version definition will be fixed in another PR until the failed dependency is resolved or another solution is figured out.

Reply:
My goal was really only to do a simple update not to spark debate to be honest. I didn't think including non supported upcoming PHP versions was a contentious point. So I'll remove this for now and we can address this in a different PR.